util/raclgen/lib.py

```diff
@@ -0,0 +1,51 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import hjson
+import sys
+from reggen.validate import check_keys
+
+# Required fields for the RACL hjson
+racl_required = {
+    'roles': ['l', 'List, specifying all RACL roles'],
+    'policies': ['g', 'Dict, specifying the policies of all RACL groups']
+}
+
+
+# Default configuration to render the RACL package for systems that don't use RACL but need the
+# type definitions
+DEFAULT_RACL_CONFIG = {
+    'nr_policies': 1,
+    'policies': {},
+}
+
+
+def parse_racl_config(config_file: str):
+    try:
+        with open(config_file, 'r') as f_racl_config:
+            racl_config = hjson.load(f_racl_config)
+    except OSError:
+        raise SystemExit(sys.exc_info()[1])
+
+    # TODO further error handling
+    error = check_keys(racl_config, racl_required, [], [], 'RACL Config')
+    if error:
+        raise SystemExit(f"Error occured while validating {config_file}")
```

Comment:
I'd strongly suggest putting a tiny bit more type checking in here. Something like extracting `roles` and `policies` as a list / dictionary, respectively (doing a basic type check) and maybe checking the item and key/value types for them respectively.

Extracting that properly will get rid of the dictionary lookups lower down in the function.

Reply:
Created https://github.com/lowRISC/opentitan/issues/25690 to track this.